eln-check.py

```diff
@@ -297,13 +301,29 @@ def diff_with_rawhide(package, eln_build=None, rawhide_build=None):
         this_package['color'] = color_none
         counter_none += 1
       package_list.append(this_package)
+    counter_total = counter_same + counter_old + counter_none
+    if counter_total == 0:
+        percentage_same = "?%"
+        percentage_old = "?%"
+        percentage_none = "?%"
+    else:
+        percentage_same = "{:.2%}".format(counter_same / counter_total)
+        percentage_old = "{:.2%}".format(counter_old / counter_total)
+        percentage_none = "{:.2%}".format(counter_none / counter_total)
     w = open(args.webpage,'w')
     w.write(tmpl.render(
       this_date = datetime.datetime.now().strftime('%Y-%m-%d %H:%M'),
       count_same = counter_same,
+      percent_same = percentage_same,
       count_old = counter_old,
+      percent_old = percentage_old,
       count_none = counter_none,
-      count_total = counter_same + counter_old + counter_none,
+      percent_none = percentage_none,
+      count_total = counter_total,
       packages = package_list
       ))
     w.close()
+
+    r = open(args.successrate,'w')
```

Comment:
The best way to close a file is by using the `with` statement. This ensures that the file is closed when the block inside the with statement is exited.

We don't need to explicitly call the close() method. It is done internally.
https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files

Reply:
I'm not sure I understand this comment saying that he needs to change his pull request.
He was following the convention of the rest of the python file.
If you feel it is best to use the with statement, don't hijack this pull request.  Please create a new one that fixes all of them.

Reply:
@sturivny Thank you for the feedback. I am aware of the convention using the `with` statement and I _almost_ implemented it that way. However, I felt it was better practice to stick with the existing style of the script.